pkg/resolver/resolver.go

```diff
@@ -284,6 +288,7 @@ func (r *resolver) Lookup(
 					// add peers to our peerstore
 					// TODO pin this in the cache
 					r.peerCache.Put(p)
+					r.blacklist.Delete(p.PublicKey().String())
```

Comment:
It would be nice to only do this if the version of the received peer is greater than the blacklisted one.

During black-listing, we could store the version as the value of the pair and check it here.
If the blacklisted version is older, we delete the blacklist.